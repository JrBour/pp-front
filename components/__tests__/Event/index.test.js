import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import Vuex from 'vuex'
import Event from '@/components/Event'
import User from '@/components/User'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(VueRouter)
const routerPush = jest.fn()
const router = new VueRouter()
router.push = routerPush

describe('EventForm', () => {
  let wrapper
  let mutations
  let store
  let state
  let addEventMock = jest.fn();

  beforeEach(() => {
    state = {
      participants: [
        {
          id: '12',
          givenName: 'John',
          lastName: 'Doe',
        }
      ],
      user: {
        id: 12
      }
    }

    mutations = {
      addEvent: addEventMock
    }
    
    store = new Vuex.Store({ state, mutations })

    wrapper = shallowMount(Event, {
      store,
      localVue,
      router,
      data: () => ({
        image: 'https://www.industrialempathy.com/img/remote/ZiClJf-640w.webp'
      }),
      stubs: {
        User,
      },
      propsData: {
        event: {
          id: '12',
          author: {
            id: 12
          },
          cover: 'image name',
          image: {
            filePath: 'gg.png'
          },
          userEvents: [
            {
              user: {
                id: 12
              }
            }
          ],
          name: 'Event name',
          description: 'Lorem ipsum',
          address: '60 rue de',
          zipcode: 90900,
          city: 'Beverly hills',
          startAt: '2013-07-02T09:00:00',
          endAt: '2013-07-02T10:00:00',
          shareFees: true
        }
      }
    })
  })

  describe('snapshot', () => {
    it('renders properly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
