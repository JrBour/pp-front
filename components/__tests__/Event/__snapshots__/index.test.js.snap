// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventForm snapshot renders properly 1`] = `
<div class="event__wrapper">
  <div class="event__header">
    <div class="event__date">
      <p class="event__date_day">Mar</p>
      <p class="event__date_date">2</p>
      <p class="event__date_month">Juil</p>
    </div>
    <div class="event__separator"></div>
    <div class="event__information">
      <h2>
        Event name
      </h2>
      <p>60 rue de - Beverly hills</p>
    </div>
    <div>
      <!----> <img src="">
    </div>
  </div>
  <div class="event__footer">
    <div class="event__participants">
      <div class="user__small">
        <div class="user__wrapper_small">
          <div class="user__image_small" style="background-image: url(); background-repeat: no-repeat; background-size: cover; background-position: center;"></div>
          <!---->
        </div>
        <!---->
      </div>
      <!---->
    </div>
    <div class="event__hours">
      <p>09h00 - 10h00</p>
    </div>
  </div>
</div>
`;
